File: frontend/src/types/simulation.ts
export interface MonthlyProduction {
  month: number;
  kwh: number;
}

export interface SimulationResult {
  id: string;
  project_id: string;
  panel_layout_id: string;
  params: Record<string, unknown>;
  monthly_production: MonthlyProduction[];
  annual_kwh: number;
  specific_yield: number | null;
  peak_power_kwc: number | null;
  performance_ratio: number | null;
  created_at: string;
}

export interface OptimizationResult {
  optimal_tilt: number;
  optimal_azimuth: number;
  annual_kwh: number;
}

File: backend/app/schemas/simulation.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class SimulateRequest(BaseModel):
    """Optional body for POST /projects/{id}/simulate."""
    panel_layout_id: uuid.UUID | None = None
    losses_pct: float = Field(default=14.0, ge=0, le=50)
    albedo: float = Field(default=0.2, ge=0, le=1.0)


class SimulationCreate(BaseModel):
    project_id: uuid.UUID
    panel_layout_id: uuid.UUID
    params: dict
    monthly_production: dict
    annual_kwh: Decimal
    specific_yield: Decimal | None = None
    peak_power_kwc: Decimal | None = None
    performance_ratio: Decimal | None = None


class SimulationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    panel_layout_id: uuid.UUID
    params: dict
    monthly_production: dict
    annual_kwh: Decimal
    specific_yield: Decimal | None = None
    peak_power_kwc: Decimal | None = None
    performance_ratio: Decimal | None = None
    created_at: datetime
